locator/ec2_snitch.cc

```diff
@@ -65,6 +66,31 @@ future<> ec2_snitch::start() {
 }
 
 future<sstring> ec2_snitch::aws_api_call(sstring addr, uint16_t port, sstring cmd, std::optional<sstring> token) {
+    int i = 0;
+    return repeat_until_value([this, addr, port, cmd, token, &i]() -> future<std::optional<sstring>> {
```

Comment:
This `&i` captures the on-stack variable that goes out-of-scope once unresolved future is returned, so manipulating `i` in continuation is unsafe. There are three options how handle it.

1. Keep `i` as `ec2_snitch::retries` variable
2. Use `seastar::do_with()`
3. Switch to coroutines

Reply:
@xemul Moved `i` to `do_with()`.